File: salesos/backend/app/modules/gtm/outreach.py
"""STORY-11-08 — CAP-103 AI Outreach models (OBJ-353-adjacent draft).

Routed through governed CAP-023 Prompt Registry — not a disconnected copy tool.
Not Production GO. DEC-085 untouched. No Alembic / FORCE RLS.
feature_ai_copilot remains False. No live SMTP / LinkedIn / WhatsApp.
"""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any


class OutreachError(ValueError):
    """Invalid AI outreach draft request or prompt binding."""


# Governed CAP-023-shaped prompt id (fixture registry — not live Studio Prompt Library).
OUTREACH_PROMPT_ID = "gtm.ai_outreach.v1"
OUTREACH_PROMPT_VERSION = "1.0.0"


@dataclass
class OutreachRequest:
    """Seed for governed outreach draft generation."""

    company_name: str
    contact_name: str = ""
    contact_title: str = ""
    channel: str = "email"
    intent: str = "intro"
    value_prop: str = ""
    website_summary: str = ""
    icp_notes: str = ""

    def as_dict(self) -> dict[str, Any]:
        return {
            "company_name": self.company_name,
            "contact_name": self.contact_name,
            "contact_title": self.contact_title,
            "channel": self.channel,
            "intent": self.intent,
            "value_prop": self.value_prop,
            "website_summary": self.website_summary,
            "icp_notes": self.icp_notes,
        }


@dataclass
class OutreachDraft:
    """Governed outreach draft (in-memory) — not a sent message."""

    id: str
    tenant_id: str
    request: OutreachRequest
    subject: str = ""
    body: str = ""
    channel: str = "email"
    prompt_id: str = OUTREACH_PROMPT_ID
    prompt_version: str = OUTREACH_PROMPT_VERSION
    spend_path: str = "platform_llm_budget"
    generator_key: str = "fixture_outreach"
    delivery_status: str = "draft_only"
    schema_version: int = 1
    created_at: str = ""
    warnings: list[str] = field(default_factory=list)

    def as_dict(self) -> dict[str, Any]:
        return {
            "id": self.id,
            "tenant_id": self.tenant_id,
            "request": self.request.as_dict(),
            "subject": self.subject,
            "body": self.body,
            "channel": self.channel,
            "prompt_id": self.prompt_id,
            "prompt_version": self.prompt_version,
            "spend_path": self.spend_path,
            "generator_key": self.generator_key,
            "delivery_status": self.delivery_status,
            "schema_version": self.schema_version,
            "created_at": self.created_at,
            "warnings": list(self.warnings),
        }


def normalize_request(
    *,
    company_name: str,
    contact_name: str | None = None,
    contact_title: str | None = None,
    channel: str | None = None,
    intent: str | None = None,
    value_prop: str | None = None,
    website_summary: str | None = None,
    icp_notes: str | None = None,
) -> OutreachRequest:
    name = (company_name or "").strip()
    if not name:
        raise OutreachError("company_name required")
    ch = (channel or "email").strip().lower() or "email"
    if ch != "email":
        raise OutreachError("only email channel supported (LinkedIn/WhatsApp deferred)")
    intent_n = (intent or "intro").strip().lower() or "intro"
    if intent_n not in {"intro", "follow_up", "breakup"}:
        raise OutreachError("intent must be intro|follow_up|breakup")
    return OutreachRequest(
        company_name=name,
        contact_name=(contact_name or "").strip(),
        contact_title=(contact_title or "").strip(),
        channel=ch,
        intent=intent_n,
        value_prop=(value_prop or "").strip()[:2000],
        website_summary=(website_summary or "").strip()[:4000],
        icp_notes=(icp_notes or "").strip()[:2000],
    )

File: salesos/frontend/src/lib/api/outreach.ts
/**
 * GTM AI Outreach HTTP (STORY-11-08 / FE-S11-08).
 * Tip FixtureOutreachGenerator + governed prompt. draft_only — no live send.
 * feature_ai_copilot remains False. Not Production GO / RAG GO.
 */
import api from "./client";

const BASE = "/api/v1/gtm/outreach";

function tenantHeaders(tenantId: string) {
  return { "X-Tenant-Id": tenantId };
}

export interface OutreachBody {
  company_name: string;
  contact_name?: string;
  contact_title?: string;
  channel?: string;
  intent?: string;
  value_prop?: string;
  website_summary?: string;
  icp_notes?: string;
  generator_key?: string;
  id?: string;
}

export interface OutreachDraft {
  id: string;
  tenant_id: string;
  request: Record<string, unknown>;
  subject: string;
  body: string;
  channel: string;
  prompt_id: string;
  prompt_version: string;
  spend_path: string;
  generator_key: string;
  delivery_status: string;
  schema_version: number;
  created_at?: string;
  warnings: string[];
}

export interface OutreachMeta {
  object: string;
  capability: string;
  prompt_id: string;
  prompt_version: string;
  channels: string[];
  intents: string[];
  spend_path: string;
  generators_configured: string[];
  delivery_status: string;
  feature_ai_copilot: boolean;
  honesty: string;
}

export async function getOutreachMeta(tenantId: string): Promise<OutreachMeta> {
  const resp = await api.get<OutreachMeta>(`${BASE}/meta`, {
    headers: tenantHeaders(tenantId),
  });
  return resp.data;
}

export async function listOutreachDrafts(tenantId: string): Promise<OutreachDraft[]> {
  const resp = await api.get<OutreachDraft[]>(BASE, {
    headers: tenantHeaders(tenantId),
  });
  return resp.data;
}

export async function getOutreachDraft(tenantId: string, runId: string): Promise<OutreachDraft> {
  const resp = await api.get<OutreachDraft>(`${BASE}/${encodeURIComponent(runId)}`, {
    headers: tenantHeaders(tenantId),
  });
  return resp.data;
}

export async function createOutreachDraft(
  tenantId: string,
  body: OutreachBody
): Promise<OutreachDraft> {
  const resp = await api.post<OutreachDraft>(BASE, body, {
    headers: tenantHeaders(tenantId),
  });
  return resp.data;
}
